### PR TITLE
feat(UrlParser): default useNewUrlParser to true

### PR DESCRIPTION
**Pull request by daprahamian**
# Description

Defaults `useNewUrlParser` to true

**What changed?**

- `lib/operations/connect.js` and `lib/operations/mongo_client_ops.js` are changed to do the following:
  - default `useNewUrlParser` to true
  - make sure to parse options in camelCase instead of lowercase.
- `lib/core/uri_parser.js` is updated to do the following:
  - respect passed in `username` and `password` options over a equivalent parsed values from the uri string
  - respect both `options.auth.(username|user|password)` and `options.(username|user|password)`

**Are there any files to ignore?**
No

** Anything else?**

There is currently one replset test error:

```
  1) Transactions spec tests read-pref - replicaset default readPreference:
     MongoError: Read preference in a transaction must be primary, not: secondary
      at applySession (lib/core/sessions.js:665:12)
      at _command (lib/core/wireprotocol/command.js:49:17)
      at Object.command (lib/core/wireprotocol/command.js:25:5)
      at cursor.topology.selectServer (lib/core/cursor.js:723:33)
      at selectServers (lib/core/sdam/topology.js:388:9)
      at selectServers (lib/core/sdam/topology.js:799:12)
      at NativeTopology.descriptionChangedHandler (lib/core/sdam/topology.js:828:7)
      at NativeTopology.EventEmitter.emit (domain.js:448:20)
      at NativeTopology.serverUpdateHandler (lib/core/sdam/topology.js:482:10)
      at Server.EventEmitter.emit (domain.js:448:20)
      at Pool.<anonymous> (lib/core/sdam/server.js:387:12)
      at Pool.EventEmitter.emit (domain.js:448:20)
      at connect (lib/core/connection/pool.js:624:10)
      at callback (lib/core/connection/connect.js:109:5)
      at runCommand (lib/core/connection/connect.js:172:5)
      at Connection.messageHandler (lib/core/connection/connect.js:334:5)
      at Connection.EventEmitter.emit (domain.js:448:20)
```

I added a commit that fixes this, and would love feedback on it.

### CODE REVIEW

File: lib/operations/connect.js [hunk 1]
```diff
@@ -586,16 +590,16 @@ function transformUrlOptions(_object) {
     object.dbName = _object.defaultDatabase;
   }
 
-  if (object.maxpoolsize) {
-    object.poolSize = object.maxpoolsize;
+  if (object.maxPoolSize) {
```
Comment (kvwalker) [inline lib/operations/connect.js]: Why are these changing to camel case?

File: lib/operations/connect.js [hunk 2]
```diff
@@ -245,6 +247,8 @@ function connect(mongoClient, url, options, callback) {
     if (_finalOptions.socketTimeoutMS == null) _finalOptions.socketTimeoutMS = 360000;
     if (_finalOptions.connectTimeoutMS == null) _finalOptions.connectTimeoutMS = 30000;
     if (_finalOptions.retryWrites == null) _finalOptions.retryWrites = true;
+    if (_finalOptions.useRecoveryToken == null) _finalOptions.useRecoveryToken = true;
+    if (_finalOptions.readPreference == null) _finalOptions.readPreference = 'primary';
```
Comment (kvwalker) [inline lib/operations/connect.js]: Would it be better to have this set in `startTransaction`? Is that where the test was failing?

Comment (daprahamian) [reply lib/operations/connect.js]: I think that might be better. My feeling here is that we might be surfacing a bug that was not present..

File: lib/operations/connect.js (hunk 1 above)
Comment (daprahamian) [reply lib/operations/connect.js]: At some point, we implementing the name normalizing logic in the new url parser that converts the values from lowercase back to camelCase. These need to be able to pull the camelCase values.

File: lib/operations/connect.js (hunk 2 above)
Comment (daprahamian) [reply lib/operations/connect.js]: Actually, looking into the specs, it says [here](https://github.com/mongodb/specifications/blob/master/source/driver-read-preferences.rst#read-preferences) that the default value for a readPreference is `primary` in a MongoClient. The reason the transaction test is failing is b/c it has no specified value, and is attempting to inherit the value from the MongoClient. So I think the current solution is proper.

**Comment by mbroadst** [comment]
@daprahamian looks like this has conflicts now unfortunately. Let's get this in first thing when you're back.

**Comment by daprahamian** [comment]
Closing in favor of #2125

**Pull request closed without merging**